### PR TITLE
PYIC-7450: update journey map to route to the two variants of the sorry-could-not-confirm-details page

### PR DESCRIPTION
**Pull request by thebauSoftwire**
### What changed
- Updated route to the two variants of the `sorry-could-not-confirm-details` page depending on the journeyContext and when `updateDetailsAccountDeletion` feature flag is enabled

### Why did it change
The `sorry-could-not-confirm-details` has two variants depending on the initial journey (rfc or reuse) which show different content so the routing needed to be updated to allow for this.

Corresponding FE PR: https://github.com/govuk-one-login/ipv-core-front/pull/1573

### Issue tracking

- [PYIC-7450](https://govukverify.atlassian.net/browse/PYIC-7450)

[PYIC-7450]: https://govukverify.atlassian.net/browse/PYIC-7450?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Joe-Edwards-GDS — COMMENTED** [review]
I think this will need to be a little bit more complicated, sorry!

If you look at the table of scenarios, we need to make sure that receiving a CI during a reuse journey also results in the new variant.

This probably means updating `CRI_STATE` (in `update-name.yaml` and `update-address.yaml`) to point CI failures (`fail-with-ci`, `enhanced-verification` and `alternate-doc-xx`) to a different place...

**Comment by Joe-Edwards-GDS** [comment]
One other thought - should we include API tests for the new routing as well? Or do you think that's covered by another ticket.

(In future I'm very keen for tickets to include API tests by default! - as you would with unit tests)

File: api-tests/features/reuse-update-details/p2-reuse-update-details-failure.feature
Comment (thebauSoftwire) [inline api-tests/features/reuse-update-details/p2-reuse-update-details-failure.feature:242]: I've only added additional failure cases here but these don't test fast-follow changes yet since there isn't a way of testing the variations in content of the page until the reinstate identity work has been merged where we'd want to test against the identity. So can wait until that work is merged or can get this in and add a separate ticket to update these api tests for fast follow?

File: lambdas/process-journey-event/src/main/resources/statemachine/journey-maps/failed.yaml
Comment (Joe-Edwards-GDS) [inline lambdas/process-journey-event/src/main/resources/statemachine/journey-maps/failed.yaml]: This is a little sneaky, but I think actually the CI events here (`fail-with-ci` etc.) should _also_ go to `COULD_NOT_CONFIRM_DETAILS_PAGE_WITH_DELETION_INVALID_ID`. And probably equivalently for the `COULD_NOT_UPDATE_DETAILS_PAGE`!

It's a bit of an unusual case, but if TICF produces a CI at this stage, it will also invalidate your existing identity...

File: api-tests/features/reuse-update-details/p2-reuse-update-details-failure.feature
Comment (Joe-Edwards-GDS) [reply api-tests/features/reuse-update-details/p2-reuse-update-details-failure.feature:242]: (Answered in slack)

**Review by Joe-Edwards-GDS — COMMENTED** [review]
Looking good, just a couple of test cases once Chris's change is merged

File: lambdas/process-journey-event/src/main/resources/statemachine/journey-maps/failed.yaml
Comment (thebauSoftwire) [reply lambdas/process-journey-event/src/main/resources/statemachine/journey-maps/failed.yaml]: good catch!

File: api-tests/features/repeat-fraud-check/p2-repeat-fraud-check-failure.feature
Comment (Joe-Edwards-GDS) [inline api-tests/features/repeat-fraud-check/p2-repeat-fraud-check-failure.feature:1]: You might find it easier to clean up later if the fast-follow versions were in a separate file - but happy to have them all together.

Either way we can tidy up after releasing the feature :)

**Review by Joe-Edwards-GDS — APPROVED** [review]
Thanks!

**Pull request merged**
